src/revisions/revisions.service.ts

```diff
@@ -101,20 +104,61 @@ export class RevisionsService {
     return revision;
   }
 
-  toRevisionMetadataDto(revision: Revision): RevisionMetadataDto {
+  async getRevisionAuthors(revision: Revision): Promise<Author[]> {
+    const authors = [];
+    for (const edit of await revision.edits) {
+      authors.push(await edit.author);
+    }
+    return authors;
+  }
+
+  async getRevisionAuthorUsernames(revision: Revision): Promise<string[]> {
+    const authors = await this.getRevisionAuthors(revision);
```

Comment:
I think we should change the implementation of this function to the implementation of the `getRevisionAuthors` function.
The current implementation is more complicated and runs the loop twice.

Reply:
```
  async getRevisionAuthorUsernames(revision: Revision): Promise<string[]> {
    const usernames: string[] = [];
    for (const edit of await revision.edits) {
      const user = await (await edit.author).user;
      if (user) {
        usernames.push(user.username);
      }
    }
    return usernames;
  }
```